sstables/index_entry.hh

```diff
@@ -541,6 +542,7 @@ public:
 
 class index_entry {
 private:
+    std::reference_wrapper<const schema> _s;
```

Comment:
Why is this a "reference_wrapper" and not a schema_ptr? Is it because (?) nothing in this code ever gets deferred? Can we be sure of that?

Reply:
I guess its needed because the class has a move assignment operator, which is incompatible with const members. We could just use a pointer instead (which is what `std::reference_wrapper<>` really is).

Reply:
_s in index reader is const& so I didn't think of using schema_ptr (isn't it less efficient?). Botond is right that I had to use reference_wrapper because of the assignment.